std/os.zig

```diff
@@ -133,6 +133,11 @@ fn getRandomBytesDevURandom(buf: []u8) !void {
     const fd = try openC(c"/dev/urandom", O_RDONLY | O_CLOEXEC, 0);
     defer close(fd);
 
+    const st = try fstat(fd);
+    if (!system.S_ISCHR(st.mode)) {
+        return error.Unexpected;
+    }
+
     const stream = &std.fs.File.openHandle(fd).inStream().stream;
     stream.readNoEof(buf) catch return error.Unexpected;
```

Comment:
Note that the reason this here is `Unexpected` is that it is in theory impossible for this situation to happen.